src/assignment/components/modals/ManageAssignmentLabels.tsx

```diff
@@ -134,12 +136,29 @@ const ManageAssignmentLabels: FunctionComponent<ManageAssignmentLabelsProps> = (
 		setAssignmentLabels([...assignmentLabels.filter((labelObj) => labelObj.id !== id)]);
 	};
 
+	const labelsExceedMaxLenght = (labels: Assignment_Label_v2[]) => {
+		let isValid = true;
+
+		labels.forEach((label) => {
+			const exceedsMaxLength = (label.label?.length || 0) > MAX_LABEL_LENGTH;
+			if (exceedsMaxLength) {
+				isValid = false;
+			}
+		});
+
+		return isValid;
+	};
+
 	const handleSaveLabels = async () => {
 		try {
 			setIsProcessing(true);
 			const initialAssignmentLabelIds = initialAssignmentLabels.map((l) => l.id);
 			const updatedAssignmentLabelIds = assignmentLabels.map((l) => l.id);
 
+			if (!labelsExceedMaxLenght(assignmentLabels)) {
+				throw new CustomError('One or more labels is longer then max length', 'invalid');
+			}
```

Comment:
preferably we don't use exceptions for handing logic we expect to happen. Since exceptions are quite heavy to throw en catch.
So if there is an easy alternative, we prefer a simple if + return;

then you can also simplify the catch statement at the end
```suggestion
			if (!labelsExceedMaxLenght(assignmentLabels)) {
				ToastService.danger(
					tHtml('Een of meerdere labels is langer dan {{maxLength}} karakters', {
						maxLength: MAX_LABEL_LENGTH,
					})
				);
				setIsProcessing(false);
				return;
			}
```